sdk/search/azure-search/src/test/java/com/azure/search/SuggestTestBase.java

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.search;
+
+import com.azure.core.http.rest.PagedResponse;
+import com.azure.search.models.SuggestResult;
+import com.azure.search.test.environment.models.Hotel;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+
+public abstract class SuggestTestBase extends SearchIndexClientTestBase {
+    static final String BOOKS_INDEX_JSON = "BooksIndexData.json";
+    static final String BOOKS_INDEX_NAME = "books";
+
+    @Override
+    protected void beforeTest() {
+        super.beforeTest();
+    }
+
+    void verifyFuzzySuggest(SuggestPagedResponse suggestResultPagedResponse) {
+        Assert.assertNotNull(suggestResultPagedResponse);
+        Assert.assertEquals(5, suggestResultPagedResponse.getValue().size());
+    }
+
+    void verifyHitHighlightingSuggest(SuggestPagedResponse suggestResultPagedResponse) {
+        Assert.assertNotNull(suggestResultPagedResponse);
+        Assert.assertEquals(1, suggestResultPagedResponse.getValue().size());
+        Assert.assertTrue(
+            suggestResultPagedResponse.getValue()
+                .get(0)
+                .getText()
+                .startsWith("Best <b>hotel</b> in town"));
+    }
+
+    void verifyFieldsExcludesFieldsSuggest(SuggestPagedResponse suggestResultPagedResponse) {
+        Assert.assertNotNull(suggestResultPagedResponse);
+        Assert.assertEquals(0, suggestResultPagedResponse.getValue().size());
+    }
+
+    void verifyDynamicDocumentSuggest(SuggestPagedResponse suggestResultPagedResponse) {
+        Assert.assertNotNull(suggestResultPagedResponse);
+        Assert.assertEquals(2, suggestResultPagedResponse.getValue().size());
+        Hotel hotel = convertToType(suggestResultPagedResponse.getValue().get(0).getDocument(), Hotel.class);
+        Assert.assertEquals("10", hotel.hotelId());
+    }
+
+    void verifyCanSuggestStaticallyTypedDocuments(SuggestPagedResponse suggestResultPagedResponse, List<Map<String, Object>> expectedHotels) {
+        //sanity
+        Assert.assertNotNull(suggestResultPagedResponse);
+        List<Document> docs = suggestResultPagedResponse.getValue()
+            .stream()
+            .map(SuggestResult::getDocument)
+            .collect(Collectors.toList());
+        List<SuggestResult> hotelsList = suggestResultPagedResponse.getValue();
+
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
@navalev This particular block of code has been repeated several times throughout the tests. To reduce the chance of introducing bugs during maintenance, it should be factored out into one place. That will also make it easier to replace it with the built-in solution once we implement it.

Reply:
@brjohnstmsft throughout the code there is work to reuse code. I've added a user story to the backlog maintenance list to continue this for the Suggest tests.